sdk/identity/identity/src/util/tracing.ts

```diff
@@ -1,54 +1,13 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { GetTokenOptions } from "@azure/core-http";
-import { getTracer, OperationTracingOptions } from "@azure/core-tracing";
-import { Span, SpanKind, SpanOptions as OTSpanOptions } from "@opentelemetry/api";
+import { createSpanFunction } from "@azure/core-tracing";
 
 /**
  * Creates a span using the global tracer.
- * @param name - The name of the operation being performed.
- * @param options - The options for the underlying http request.
+ * @internal
  */
-export function createSpan(
-  operationName: string,
-  options: GetTokenOptions = {}
-): { span: Span; options: GetTokenOptions } {
-  const tracer = getTracer();
-
-  const tracingOptions: OperationTracingOptions = {
-    spanOptions: {},
-    ...options.tracingOptions
-  };
-
-  const spanOptions: OTSpanOptions = {
-    ...tracingOptions.spanOptions,
-    kind: SpanKind.INTERNAL
-  };
-
-  const span = tracer.startSpan(`Azure.Identity.${operationName}`, spanOptions);
-  span.setAttribute("az.namespace", "Microsoft.AAD");
-
-  let newOptions = options;
-  if (span.isRecording()) {
-    newOptions = {
-      ...options,
-      tracingOptions: {
-        ...tracingOptions,
-        spanOptions: {
-          ...tracingOptions.spanOptions,
-          parent: span.context(),
-          attributes: {
-            ...spanOptions.attributes,
-            "az.namespace": "Microsoft.AAD"
-          }
-        }
-      }
-    };
-  }
-
-  return {
-    span,
-    options: newOptions
-  };
-}
+export const createSpan = createSpanFunction({
+  packagePrefix: "Azure.Identity",
```

Comment:
@sadasant  - can you review this and see if looks okay? This is a smaller PR than we originally discussed - it's just centralizing everyone on using the same (and single) createSpan implementation from core-tracing.

Reply:
Who doesn't like some curry(ing)? 🌞 This is miles better!

Reply:
Also, thank you for making a smaller PR. Much better than a larger PR!

Reply:
Yeah, the secret sauce was "do this PR before I do the actual otel upgrade". There will still be some fallout changes from that, but for the most part they're mitigated so long as everyone treats the opentelemetry models as opaque (which once this is in, they will!)